### PR TITLE
[DEV-1066] Webinars section

### PR DESCRIPTION
**Pull request by marcobottaro**
#### List of Changes

Add webinars section with mocked data

#### How Has This Been Tested?

Manually

#### Screenshots (if appropriate):
image

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: apps/nextjs-website/src/components/organisms/WebinarsSection/WebinarsSection.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+import { Box, Typography, useTheme } from '@mui/material';
+import React from 'react';
+import WebinarCard from '@/components/molecules/WebinarCard/WebinarCard';
+import LinkButton from '@/components/atoms/LinkButton/LinkButton';
+import EContainer from '@/editorialComponents/EContainer/EContainer';
+
+export type webinarsSectionProps = {
+  title: string;
+  description: string;
+  link: { href?: string; label: string };
```
Comment (MarcoPonchia) [inline apps/nextjs-website/src/components/organisms/WebinarsSection/WebinarsSection.tsx:12]: This field should be optional
image

```suggestion
  link?: { href?: string; label: string };
```

File: apps/nextjs-website/src/app/page.tsx [hunk 2]
```diff
@@ -35,6 +38,12 @@ const Home = async () => {
           svgPath: product.svgPath,
         }))}
       />
+      <WebinarsSection
+        title={homepage.webinarsSection.title}
+        description={homepage.webinarsSection.description}
+        link={homepage.webinarsSection.link}
```
Comment (MarcoPonchia) [inline apps/nextjs-website/src/app/page.tsx:44]: I think is better to remove link for now that webinars list page does not esist yet
```suggestion
```

**Review by MarcoPonchia — CHANGES_REQUESTED** [review]

File: apps/nextjs-website/src/components/molecules/WebinarCard/WebinarCard.tsx [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+          />
+          <Box mt={4}>
+            <ButtonNaked
+              component={Link}
+              href={path}
+              color={'primary'}
+              variant={'contained'}
+            >
+              {webinar.subscribe}
+            </ButtonNaked>
```
Comment (MarcoPonchia) [inline apps/nextjs-website/src/components/molecules/WebinarCard/WebinarCard.tsx:67]: This should be a component coming from props

File: apps/nextjs-website/src/lib/api.ts [hunk 4]
```diff
@@ -128,3 +131,14 @@ export async function getTutorialLists(productSlug?: string) {
     ) || null;
   return manageUndefinedAndAddProduct(props);
 }
+
+export async function getWebinars(): Promise<readonly Webinar[]> {
+  // TODO: Remove this condition when the webinar content is ready
+  return isProduction ? [] : [...webinars];
```
Comment (kin0992) [inline apps/nextjs-website/src/lib/api.ts:137]:
```suggestion
  return isProduction ? [] : webinars;
```

File: apps/nextjs-website/src/lib/api.ts [hunk 5]
```diff
@@ -128,3 +131,14 @@ export async function getTutorialLists(productSlug?: string) {
...
+}
+
+export async function getNextWebinars(): Promise<readonly Webinar[]> {
+  return (await getWebinars()).filter(
+    ({ startDateTime }) => startDateTime && startDateTime > new Date()
```
Comment (kin0992) [inline apps/nextjs-website/src/lib/api.ts:142]:
```suggestion
    ({ startDateTime }) => startDateTime && startDateTime.getTime() > new Date().getTime()
```
in this case you are comparing numbers, starting from epoch, which is safer

File: apps/nextjs-website/src/lib/api.ts [hunk 6]
```diff
@@ -128,3 +131,14 @@ export async function getTutorialLists(productSlug?: string) {
     ) || null;
   return manageUndefinedAndAddProduct(props);
 }
+
+export async function getWebinars(): Promise<readonly Webinar[]> {
```
Comment (kin0992) [inline apps/nextjs-website/src/lib/api.ts:135]:
```suggestion
async function getWebinars(): Promise<readonly Webinar[]> {
```

Comment (kin0992) [inline apps/nextjs-website/src/lib/api.ts:135]: you could even use a single function that accept an optional filter parameter.
e.g. `getWebinars(filter: FilterType) => ...` where filter type could be anything we want, like (`{ startDate: Date, ... }`).
Maybe we can change this later

File: apps/nextjs-website/src/components/atoms/TimeSlot/TimeSlot.tsx [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+  day: 'numeric',
+};
+
+const TIME_OPTIONS: Intl.DateTimeFormatOptions = {
+  hour: '2-digit',
+  minute: '2-digit',
+};
+
+function formattedDateTime(date: Date): string {
+  return `${new Date(date).toLocaleDateString(
```
Comment (kin0992) [inline apps/nextjs-website/src/components/atoms/TimeSlot/TimeSlot.tsx:17]: why `new Date(date).toLocaleDateString` and not only `date.toLocaleDateString`?

File: apps/nextjs-website/src/components/atoms/TimeSlot/TimeSlot.tsx [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+  month: 'long',
+  day: 'numeric',
+};
+
+const TIME_OPTIONS: Intl.DateTimeFormatOptions = {
+  hour: '2-digit',
+  minute: '2-digit',
+};
+
+function formattedDateTime(date: Date): string {
```
Comment (kin0992) [inline apps/nextjs-website/src/components/atoms/TimeSlot/TimeSlot.tsx:16]: could you add a comment on this function?

File: apps/nextjs-website/src/components/atoms/TimeSlot/TimeSlot.tsx [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+  return `${new Date(date).toLocaleDateString(
+    DEFAULT_LOCALE,
+    DATE_OPTIONS
+  )}, ${new Date(date).toLocaleTimeString(DEFAULT_LOCALE, TIME_OPTIONS)}`;
+}
+
+function conditionallyFormattedEndDate(
```
Comment (kin0992) [inline apps/nextjs-website/src/components/atoms/TimeSlot/TimeSlot.tsx:23]: could you add a comment on this function?

File: apps/nextjs-website/src/lib/api.ts (hunk 6 above)
Comment (marcobottaro) [reply apps/nextjs-website/src/lib/api.ts:135]: Yes, we will change all these functions once the logic about webinars is established

**Review by MarcoPonchia — APPROVED** [review]

File: apps/nextjs-website/src/app/page.tsx [hunk 10]
```diff
@@ -35,6 +37,13 @@ const Home = async () => {
           svgPath: product.svgPath,
         }))}
       />
+      {nextWebinars.length && (
```
Comment (kin0992) [inline apps/nextjs-website/src/app/page.tsx:40]:
```suggestion
      {nextWebinars.length !== 0 && (
```
without that, with the data you are passing right now, you will see a `0` rendered

**Review by kin0992 — APPROVED** [review]

**Pull request merged**